### PR TITLE
fix #691

### LINKED ISSUE #691: KDB_DB_SPEC and KDB_DB_SYSTEM with ~ does not work
SPEC Expected behaviour: Should resolve to `~/.config/kdb/spec/default.ini`
SPEC Actual behaviour: Resolves to `/usr/local/~/.config/kdb/spec/default.ini`

SYSTEM Expected behaviour: Should resolve to `/home/me/...`
SYSTEM actual behaviour: resolves to `~/...`

KDB_DB_SYSTEM also needs to be STRING, otherwise it is impossible to start string with `~`.

### PR DESCRIPTION
**Pull request by tom-wa**

### CODE REVIEW

File: src/plugins/resolver/filename.c [hunk 1]
```diff
@@ -406,11 +422,26 @@ static int elektraResolveSpec (resolverHandle * p, Key * warningsKey ELEKTRA_UNU
 	}
 	else
 	{
-		strcpy (p->filename, KDB_DB_SPEC);
-		strcat (p->filename, "/");
-		strcat (p->filename, p->path);
+		if (KDB_DB_SPEC[0] == '~')
```
Comment (markus2330) [inline src/plugins/resolver/filename.c]: Did you already consider discussion at #691? ~ would not be on the first position.

Comment (tom-wa) [inline src/plugins/resolver/filename.c]: not really, this version would only work after dropping CMAKE_INSTALL_PREFIX for pathnames starting with `~`
i don't really see an other way of doing it inside the resolver

Comment (markus2330) [inline src/plugins/resolver/filename.c]: You need to add a new variable in kdbconfig.h, maybe KDB_DB_SPEC_RELATIVE.

**Comment by markus2330** [comment]
Please update as discussed.

**Comment by markus2330** [comment]
@tom-wa please ping if finished

**Comment by tom-wa** [comment]
@markus2330 done

**Comment by markus2330** [comment]
thank you!

**Pull request merged**